lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html

```diff
@@ -78,7 +78,7 @@
                 </adf-categories-management>
             </div>
         </ng-container>
-        <mat-expansion-panel *ngFor="let customPanel of customPanels">
+        <mat-expansion-panel *ngFor="let customPanel of customPanels" [expanded]="canExpandTheCard(customPanel.panelTitle)">
```

Comment:
Can we not use function call here? Something like
```
[expanded]="customPanel.panelTitle === displayAspect"
```
Although since this is an expression here instead of a fixed variable, I'm not sure if it will actually have any impact or not. @AleksanderSklorz what would you suggest?

Reply:
Maybe we can do something like add additional field in CustomPanel and that field would be set when 'displayAspect' input is changed because canExpandTheCard depends on displayAspect. 
So then we can refer to CustomPanel's field instead of calling function.

Reply:
The only problem that I can think of with this approach is the fact that by adding `expanded` property to the custom panel we kind of allow other programmers to change its state outside of the metadata component which should only be possible by changing `displayAspect` value @AleksanderSklorz what do you think?

Reply:
I see your point @MichalKinas. Only idea which I see is having object like: 
someObject: { [key: string]: boolean } = {}; 

in your component and then fill this object with pairs when at the left side will be something what identifies panel (like panelTitle) and at right side boolean if it is expanded. 

So then you can fill that object in your component and it can't be changed from parent component + you can then read this without calling functions in HTML like: 

[expanded]="someObject[customPanel.panelTitle]"

Reply:
So in that way you will get this value like by referring to just field instead of by function.

Reply:
I was just wondering which solution would be most effective in this case, to achieve the solution you proposed we would have to add custom setter to `customPanels`, `grouppedProperties` and `displayAspect` to make sure we always get correct panel to expand which will add additional operations and complexity to the code. And in this case the function we're calling executes only one string comparison so it is not really expensive that is why I wonder what do you guys think? @AleksanderSklorz @swapnil-verma-gl

Reply:
Yeah good explanation. This function is not very complex so we can leave it but let's see what @swapnil-verma-gl says

Reply:
I'm fine with it. If I remember correctly, it was decided to leave the instances where moving from a function call to a fixed variable proved to be too complex. As you've already explained, updating the code would add too much complexity, so we can leave it

Reply:
I'm closing my comments then

Reply:
Thanks guys!

Reply:
Marking my original comment as resolved. Thanks!!!